usr/share/rear/output/default/94_grub2_rescue.sh

```diff
@@ -1,146 +1,164 @@
+password_pbkdf2 $GRUB_SUPERUSER $grub_rescue_password_PBKDF2_hash
 EOF" > /etc/grub.d/01_users
 fi
 
-grub_pass_set=$(tail -n 4 /etc/grub.d/01_users | grep -E "cat|set superusers|password_pbkdf2|EOF" | wc -l)
-if [[ $grub_pass_set < 4 ]]; then
+grub_pass_set=$( tail -n 4 /etc/grub.d/01_users | grep -E "cat|set superusers|password_pbkdf2|EOF" | wc -l )
+if [[ $grub_pass_set < 4 ]] ; then
     echo "#!/bin/sh
 cat << EOF
 set superusers=\"$GRUB_SUPERUSER\"
-password_pbkdf2 $GRUB_SUPERUSER $GRUB_RESCUE_PASSWORD
+password_pbkdf2 $GRUB_SUPERUSER $grub_rescue_password_PBKDF2_hash
 EOF" > /etc/grub.d/01_users
 fi
 
-grub_super_set=$(grep 'set superusers' /etc/grub.d/01_users | cut -f2 -d '"')
-if [[ ! $grub_super_set == $GRUB_SUPERUSER ]]; then
+grub_super_set=$( grep 'set superusers' /etc/grub.d/01_users | cut -f2 -d '"' )
+if [[ ! $grub_super_set == $GRUB_SUPERUSER ]] ; then
     sed -i "s/set superusers=\"\S*\"/set superusers=\"$GRUB_SUPERUSER\"/" /etc/grub.d/01_users
-    sed -i "s/password_pbkdf2\s\S*\s\S*/password_pbkdf2 $GRUB_SUPERUSER $GRUB_RESCUE_PASSWORD/" /etc/grub.d/01_users
+    sed -i "s/password_pbkdf2\s\S*\s\S*/password_pbkdf2 $GRUB_SUPERUSER $grub_rescue_password_PBKDF2_hash/" /etc/grub.d/01_users
 fi
 
-grub_enc_password=$(grep "password_pbkdf2" /etc/grub.d/01_users | awk '{print $3}')
-if [[ ! $grub_enc_password == $GRUB_RESCUE_PASSWORD ]]; then
-    sed -i "s/password_pbkdf2\s\S*\s\S*/password_pbkdf2 $GRUB_SUPERUSER $GRUB_RESCUE_PASSWORD/" /etc/grub.d/01_users
+grub_enc_password=$( grep "password_pbkdf2" /etc/grub.d/01_users | awk '{print $3}' )
+if [[ ! $grub_enc_password == $grub_rescue_password_PBKDF2_hash ]] ; then
+    sed -i "s/password_pbkdf2\s\S*\s\S*/password_pbkdf2 $GRUB_SUPERUSER $grub_rescue_password_PBKDF2_hash/" /etc/grub.d/01_users
 fi
 
 # Ensure 01_users is added to the /boot/grub.d/
-if [[ ! -x /etc/grub.d/01_users ]]; then
+if [[ ! -x /etc/grub.d/01_users ]] ; then
     chmod 755 /etc/grub.d/01_users
 fi
 
-#Finding UUID of filesystem containing /boot
-grub_boot_uuid=$(df /boot | awk 'END {print $1}' | xargs blkid -s UUID -o value)
+# Finding UUID of filesystem containing /boot
+grub_boot_uuid=$( df /boot | awk 'END {print $1}' | xargs blkid -s UUID -o value )
 
-#Stop if $grub_boot_uuid is not a valid UUID
-blkid -U $grub_boot_uuid > /dev/null 2>&1
-StopIfError "$grub_boot_uuid is not a valid UUID"
+# Stop if $grub_boot_uuid is not a valid UUID
+blkid -U $grub_boot_uuid > /dev/null 2>&1 || Error "$grub_boot_uuid is not a valid UUID"
 
-#Creating REAR grub menu entry
+# Creating Relax-and-Recover grub menu entry
 echo "#!/bin/bash
 cat << EOF
-menuentry \"Relax and Recover\" --class os --users \"\" {
+menuentry \"Relax-and-Recover\" --class os --users \"\" {
         search --no-floppy --fs-uuid  --set=root $grub_boot_uuid
-        linux  /rear-kernel $KERNEL_CMDLINE
-        initrd /rear-initrd.cgz
-        password_pbkdf2 $GRUB_SUPERUSER $GRUB_RESCUE_PASSWORD
+        linux  /boot/rear-kernel $KERNEL_CMDLINE
```

Comment:
@jsmeix are you sure `/boot` is required? Does the `--set=root` not fulfil this? Just want to double check (I could be wrong).